pkg/apis/gke.cattle.io/v1/types.go

```diff
@@ -89,7 +89,7 @@ type ClusterAddons struct {
 
 type NodePoolConfig struct {
 	Autoscaling       *NodePoolAutoscaling `json:"autoscaling,omitempty"`
-	Config            *NodeConfig          `json:"config,omitempty"`
+	Config            *GKENodeConfig       `json:"config,omitempty"`
```

Comment:
I'm slightly concerned about this case affecting other items in API. If we add a all-cluster-wide `NodePoolAutoscaling` in future that would clash with this one (potentially silently).

Reply:
Yes I think we should change at least the autoscaling one. Probably the management one too, even though thats more obscure, it uses very generic phrasing.
Is this conflicting only at the nodePoolConfig level or at all levels ? Meaning could `NodeTaintConfig` or `CidrBlock` below do same thing ?

Reply:
I have not observed any other conflict aside from https://github.com/rancher/gke-operator/pull/8. It's not specific to the `nodePoolConfig` level. Anything defined here that causes `go generate` in rancher to generate a client schema has the potential to conflict with something in rancher, at any level. That doesn't mean it will. I'd rather either fix things on a case-by-case basis like I've done here, or preemptively fix everything, but I would prefer not to pick and choose at random. `NodePoolAutoscaling` isn't necessarily more or less likely to cause a conflict than `IPAllocationPolicy` or `CidrBlock`.

Reply:
btw in the case of https://github.com/rancher/gke-operator/pull/8 it was very obvious because rancher would not build because `go generate` wouldn't overwrite the `networkConfig` generated file. This case is more sneaky because the conflict was a dynamic schema and I think it's less likely there will be many of those.

Reply:
Well, doesn't hurt anything to move them all to being GKE specific. Seems worth it since there is no real tradeoff other than minor readability.

Reply:
Renamed them all